config/confighttp/confighttp.go

```diff
@@ -58,6 +58,21 @@ type HTTPClientSettings struct {
 
 	// Auth configuration for outgoing HTTP calls.
 	Auth *configauth.Authentication `mapstructure:"auth,omitempty"`
+
+	// MaxIdleConns is used to set a limit to the maximum idle HTTP connections the client can keep open.
+	// Here pointer is used to differentiate `no input` from `zero value input`
+	MaxIdleConns *int `mapstructure:"max_idle_conns"`
+
+	// MaxIdleConnsPerHost is used to set a limit to the maximum idle HTTP connections the host can keep open.
+	MaxIdleConnsPerHost int `mapstructure:"max_idle_conns_per_host"`
+
+	// MaxConnsPerHost limits the total number of connections per host, including connections in the dialing,
+	// active, and idle states.
+	MaxConnsPerHost int `mapstructure:"max_conns_per_host"`
+
+	// IdleConnTimeout is the maximum amount of time a connection will remain open before closing itself.
+	// Here pointer is used to differentiate `no input` from `zero value input`
```

Comment:
It would be nice to explain more details, e.g. that there's an already set value and we want to override it only if an explicit value exists, etc.

Reply:
Done